ocaml/xapi/cli_frontend.ml

```diff
@@ -2066,6 +2066,22 @@ add a mapping of 'path' -> '/tmp', the command line should contain the argument
       implementation=No_fd Cli_operations.net_attach;
       flags=[Hidden];
     };
+    "network-add-purpose",
+    {
+      reqd=["uuid"; "network-purpose"];
```

Comment:
I think `purpose` is definitely better here, because it's shorter, and avoids duplication, we already know that this is a network.

Reply:
Using just `purpose` would work fine for now, but I did it this way because the bash autocompletion mechanism is based on parameter name.
If in future we have a purpose field for other objects, we'll want the autocompletion to use the appropriate purpose-list for that `thing-purpose` where `thing<>network`.

Reply:
So we only know the parameter name in the completion script but not the whole command? That's sad :-(.

Reply:
For most params it makes sense to re-use the expansion across commands, e.g. network-uuid, power-state etc.